# 01 Importing Libraries

In [2]:
# Import libraries
import pandas as pd
import numpy as np
import os

df = pd.read_csv(r'/Users/Dom/Documents/Instacart Basket Analysis/02 Data/Original Data/orders.csv',index_col = False)

In [10]:
dfsampleimp = pd.read_csv(r'/Users/Dom/Documents/Instacart Basket Analysis/03 Scripts/sample.csv',index_col = False)

In [11]:
dfsampleimp.shape

(10000, 8)

In [12]:
dfsampleimp.head(30)

Unnamed: 0  order_id  user_id eval_set  order_number  order_dow  \
0      3098739   3370853   186861    prior             8          5   
1       189466   2054041    11457    prior            17          1   
2      2414520    433512   145285    prior            30          1   
3       498382    765069    30056    prior            72          3   
4       884731   3278091    53153    prior             7          2   
5      3400469   2241880   204930    prior             8          2   
6        53253   2350754     3310    prior             2          2   
7      1000802   2061339    60134    prior            10          4   
8      3256063   3286021   196243    prior             5          0   
9      3148396   1177150   189851    prior            21          3   
10      237650    400354    14412    prior            21          0   
11      457983   1649749    27574    prior            35          5   
12     2031655   1457461   122048    prior             9          6   
13      110490   2239495     6685    prior             4          4   
14     2665045   1737417   160426    prior            84          5   
15     1134580   2204742    68214    prior             5          0   
16       40317   1165160     2488    prior             4          1   
17     1313200   2848381    78922    prior            69          5   
18     3340748    579103   201377    prior             1          4   
19     1332308   1910180    80049    prior             5          0   
20        2798   2096989      187    prior             4          1   
21      220730   1530994    13335    prior            15          4   
22     1403504   1238308    84321    prior            83          4   
23      547533   1266182    32994    prior            49          1   
24      356322   1028016    21535    prior            39          1   
25     1502259   1865838    90280    prior             5          4   
26      437359   1324683    26284    prior            12          0   
27     1240520   2132464    74580    prior             1          3   
28     2271483   2389564   136775    prior             8          3   
29     2634127    412162   158594    prior            31          0   

    order_hour_of_day  days_since_prior_order  
0                  13                    30.0  
1                  12                     2.0  
2                  11                     7.0  
3                  22                     1.0  
4                  16                    30.0  
5                  14                    10.0  
6                  11                     8.0  
7                  15                     5.0  
8                   7                     8.0  
9                  16                     9.0  
10                 13                     6.0  
11                 15                     9.0  
12                 15                     0.0  
13                 12                    30.0  
14                 14                     3.0  
15                 16                     8.0  
16                 10                    11.0  
17                 19                     1.0  
18                 20                     NaN  
19                 15                     4.0  
20                 17                     3.0  
21                 12                     3.0  
22                  8                     7.0  
23                  8                     4.0  
24                 12                     5.0  
25                 19                    11.0  
26                 15                    17.0  
27                 13                     NaN  
28                 10                     4.0  
29                 10                    14.0

In [16]:
df.tail()

order_id  user_id eval_set  order_number  order_dow  \
3421078   2266710   206209    prior            10          5   
3421079   1854736   206209    prior            11          4   
3421080    626363   206209    prior            12          1   
3421081   2977660   206209    prior            13          1   
3421082    272231   206209    train            14          6   

         order_hour_of_day  days_since_prior_order  
3421078                 18                    29.0  
3421079                 10                    30.0  
3421080                 12                    18.0  
3421081                 12                     7.0  
3421082                 14                    30.0

In [13]:
dfsampleimp.shape

(10000, 8)

In [14]:
dfsampleimp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 8 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Unnamed: 0              10000 non-null  int64  
 1   order_id                10000 non-null  int64  
 2   user_id                 10000 non-null  int64  
 3   eval_set                10000 non-null  object 
 4   order_number            10000 non-null  int64  
 5   order_dow               10000 non-null  int64  
 6   order_hour_of_day       10000 non-null  int64  
 7   days_since_prior_order  9354 non-null   float64
dtypes: float64(1), int64(6), object(1)
memory usage: 625.1+ KB


In [15]:
dfsampleimp.dtypes

Unnamed: 0                  int64
order_id                    int64
user_id                     int64
eval_set                   object
order_number                int64
order_dow                   int64
order_hour_of_day           int64
days_since_prior_order    float64
dtype: object

In [16]:
dfsampleimp.columns

Index(['Unnamed: 0', 'order_id', 'user_id', 'eval_set', 'order_number',
       'order_dow', 'order_hour_of_day', 'days_since_prior_order'],
      dtype='object')

In [17]:
dfsampleimp.describe()

Unnamed: 0      order_id       user_id  order_number     order_dow  \
count  1.000000e+04  1.000000e+04   10000.00000   10000.00000  10000.000000   
mean   1.704034e+06  1.710455e+06  102587.13680      17.04020      2.747800   
std    9.887724e+05  9.914211e+05   59604.60914      17.79187      2.040346   
min    3.310000e+02  4.700000e+01      27.00000       1.00000      0.000000   
25%    8.470942e+05  8.519430e+05   50903.50000       5.00000      1.000000   
50%    1.706401e+06  1.700560e+06  102446.50000      11.00000      3.000000   
75%    2.568720e+06  2.567440e+06  154558.25000      23.00000      5.000000   
max    3.420325e+06  3.420912e+06  206162.00000     100.00000      6.000000   

       order_hour_of_day  days_since_prior_order  
count       10000.000000             9354.000000  
mean           13.521500               11.120056  
std             4.209003                9.184078  
min             0.000000                0.000000  
25%            10.000000                4.000000  
50%            13.000000                7.000000  
75%            16.000000               16.000000  
max            23.000000               30.000000

In [18]:
path = r'/Users/Dom/Documents/Instacart Basket Analysis'

In [19]:
df = pd.read_csv(os.path.join(path, '03 Scripts', 'sample.csv'), nrows = 10000)

In [20]:
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order']

In [21]:
df = pd.read_csv(os.path.join(path, '03 Scripts', 'sample.csv'), usecols = vars_list)

In [22]:
## Step 6 - Print First 5 Rows of DataFrame
df.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   3370853   186861             8          5                 13   
1   2054041    11457            17          1                 12   
2    433512   145285            30          1                 11   
3    765069    30056            72          3                 22   
4   3278091    53153             7          2                 16   

   days_since_prior_order  
0                    30.0  
1                     2.0  
2                     7.0  
3                     1.0  
4                    30.0

In [24]:
## Step 7
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'products.csv'), index_col = False)
df_prods.shape

(49693, 5)

In [76]:
## Step 8 - Print the first 20 rows & last 35 Rows

df_prods.head(20)

product_id                                       product_name  aisle_id  \
0            1                         Chocolate Sandwich Cookies        61   
1            2                                   All-Seasons Salt       104   
2            3               Robust Golden Unsweetened Oolong Tea        94   
3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4            5                          Green Chile Anytime Sauce         5   
5            6                                       Dry Nose Oil        11   
6            7                     Pure Coconut Water With Orange        98   
7            8                  Cut Russet Potatoes Steam N' Mash       116   
8            9                  Light Strawberry Blueberry Yogurt       120   
9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
10          11                                  Peach Mango Juice        31   
11          12                         Chocolate Fudge Layer Cake       119   
12          13                                  Saline Nasal Mist        11   
13          14                     Fresh Scent Dishwasher Cleaner        74   
14          15                           Overnight Diapers Size 6        56   
15          16                      Mint Chocolate Flavored Syrup       103   
16          17                                  Rendered Duck Fat        35   
17          18                Pizza for One Suprema  Frozen Pizza        79   
18          19   Gluten Free Quinoa Three Cheese & Mushroom Blend        63   
19          20     Pomegranate Cranberry & Aloe Vera Enrich Drink        98   

    department_id  prices  
0              19     5.8  
1              13     9.3  
2               7     4.5  
3               1    10.5  
4              13     4.3  
5              11     2.6  
6               7     4.4  
7               1     1.1  
8              16     7.0  
9               7     8.4  
10              7     2.8  
11              1     9.4  
12             11     1.1  
13             17     6.5  
14             18    11.2  
15             19     5.2  
16             12    17.1  
17              1    10.6  
18              9    12.0  
19              7     6.0

In [77]:
df_prods.tail(35)

product_id                                       product_name  \
49658       49654                   Teriyaki Sauce, Sesame, Original   
49659       49655                                        Apple Cider   
49660       49656                         Masada Kosher Pocket Bread   
49661       49657                                  Cabernet Tomatoes   
49662       49658                         Brie with Herbs Foil Wedge   
49663       49659                            Organic Creamed Coconut   
49664       49660                        Professionals Sleek Shampoo   
49665       49661                                              Porto   
49666       49662                       Bacon Cheddar Pretzel Pieces   
49667       49663  Ultra Protein Power Crunch Peanut Butter N' Ho...   
49668       49664                     Lemon Cayenne Drinking Vinegar   
49669       49665      Super Dark Coconut Ash & Banana Chocolate Bar   
49670       49666                      Ginger Snaps Snacking Cookies   
49671       49667           Enchilada with Spanish Rice & Beans Meal   
49672       49668                     Apple Cinnamon Scented Candles   
49673       49669                                   K Cup Dark Blend   
49674       49670                                Beef Summer Sausage   
49675       49671                               Milk Chocolate Drops   
49676       49672                             Cafe Mocha K-Cup Packs   
49677       49673              Stone Baked Multi Grain Artisan Rolls   
49678       49674            Frozen Greek Yogurt Bars Chocolate Chip   
49679       49675                Cinnamon Dolce Keurig Brewed K Cups   
49680       49676                             Ultra Red Energy Drink   
49681       49677                    Thick & Chunky Sloppy Joe Sauce   
49682       49678                    Large Chicken & Cheese Taquitos   
49683       49679                            Famous Chocolate Wafers   
49684       49680                 All Natural Creamy Caesar Dressing   
49685       49681            Spaghetti with Meatballs and Sauce Meal   
49686       49682                                 California Limeade   
49687       49683                                     Cucumber Kirby   
49688       49684          Vodka, Triple Distilled, Twist of Vanilla   
49689       49685                 En Croute Roast Hazelnut Cranberry   
49690       49686                                   Artisan Baguette   
49691       49687         Smartblend Healthy Metabolism Dry Cat Food   
49692       49688                             Fresh Foaming Cleanser   

       aisle_id  department_id  prices  
49658         5             13     4.0  
49659        98              7    10.7  
49660       128              3     7.1  
49661        83              4     8.3  
49662         2             16     3.9  
49663        17             13     3.1  
49664        22             11     6.7  
49665       134              5     8.2  
49666       107             19     3.6  
49667        57             14    10.2  
49668       100             21    13.7  
49669        45             19     6.9  
49670        61             19     5.2  
49671        38              1     6.6  
49672       101             17     5.6  
49673       100             21     4.7  
49674       106             12    19.2  
49675        45             19     3.0  
49676        26              7     6.5  
49677       129              1     5.6  
49678        37              1    11.1  
49679        26              7    14.0  
49680        64              7    14.5  
49681        59             15     8.9  
49682       129              1     3.4  
49683        61             19     6.0  
49684        89             13     4.9  
49685        38              1     6.9  
49686        98              7     4.3  
49687        83              4    13.2  
49688       124              5     5.3  
49689        42              1     3.1  
49690       112              3     7.8  
49691        41     

In [78]:
## Step 9

df_prods.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [79]:
## Step10

df_prods.shape

(49693, 5)

In [81]:
df_prods.dtypes

product_id         int64
product_name      object
aisle_id           int64
department_id      int64
prices           float64
dtype: object

In [83]:
df_prods.describe()

product_id      aisle_id  department_id        prices
count  49693.000000  49693.000000   49693.000000  49693.000000
mean   24844.345139     67.770249      11.728433      9.994136
std    14343.717401     38.316774       5.850282    453.519686
min        1.000000      1.000000       1.000000      1.000000
25%    12423.000000     35.000000       7.000000      4.100000
50%    24845.000000     69.000000      13.000000      7.100000
75%    37265.000000    100.000000      17.000000     11.200000
max    49688.000000    134.000000      21.000000  99999.000000

# Step 11 Answer
What is the maximum value of the "aisle_id" column? The answer: 134
What is the data type of the "department_id" column? Answer: integer